src/server.jl

```diff
@@ -706,16 +719,45 @@ function evaluate_raw_cells!(
                             end
                         end
                     end
+
+                    cell_options = expand_cell ? remote.cell_options : Dict()
+
+                    if chunk.language === :r
+                        # Code cells always get the language of the notebook assigned, in this case julia,
+                        # so to render an R formatted cell, we need to do a workaround. We push a cell before
+                        # the actual code cell which contains a plain markdown block that wraps the code in ```r
+                        # for the formatting.
+                        push!(
+                            cells,
+                            (;
+                                id = string(
+                                    expand_cell ? string(nth, "_", mth) : string(nth),
+                                    "_code_prefix",
+                                ),
+                                cell_type = :markdown,
+                                metadata = (;),
+                                source = process_cell_source(
+                                    """
+           ```r
+           $(strip_cell_options(chunk.source))
+           ```
+           """,
+                                    Dict(),
+                                ),
+                            ),
+                        )
+                        # We also need to hide the real code cell in this case, which contains possible formatting
+                        # settings in its YAML front-matter and which can therefore not be omitted entirely.
+                        cell_options["echo"] = false
+                    end
+
                     push!(
                         cells,
                         (;
                             id = expand_cell ? string(nth, "_", mth) : string(nth),
                             cell_type = chunk.type,
                             metadata = (;),
-                            source = process_cell_source(
-                                remote.code,
-                                expand_cell ? remote.cell_options : Dict(),
-                            ),
+                            source = process_cell_source(chunk.source, cell_options),
```

Comment:
A couple lines below here is the code that handles inline code evaluation, e.g.

```qmd
Some inline code: `{julia} 1+1`.
```

Should we also be supporting `{r}` here as well?

Reply:
Good idea